src/parsers/trades/liquid/index.ts

```diff
@@ -0,0 +1,175 @@
+                        break;
+                    }
+                    case 6: {
+                        internalFormat.push(addFeeTrade(tradeToAdd, trades[4], trades[5]));
+                        break;
+                    }
+                    case 8: {
+                        let secondTrade = tradeToAdd;
+                        internalFormat.push(addFeeTrade(tradeToAdd, trades[4], trades[5]));
+                        internalFormat.push(addFeeTrade(secondTrade, trades[6], trades[7]));
+                        break;
+                    }
+                    default: {
+                        console.warn(`Error parsing ${tradeToAdd.exchange} trade. It extends over ${trades.length} lines`);
+                        console.info(trades);
+                    }
+                }
+                break;
+            }
+            case 'rebate_trade_fee':
+            case 'trade_fee': {
+                console.error(`Error parsing ${tradeToAdd.exchange} trade. First line should not be of type ${trades[0].transaction_type}`);
+                break;
+            }
+            default: {
+                console.log(`Ignored ${tradeToAdd.exchange} trade of type ${trades[0].transaction_type}`);
+            }
+        }
+    }
+    return internalFormat;
+}
+
+function addSingleLineTrade(
+    tradeToAdd: IPartialTrade,
+    trade: ILiquid,
+) : ITrade {
+    if (trade.direction.toUpperCase() === LiquidOrderDirection.PAY) {
+        tradeToAdd.boughtCurrency = 'USDT';
+        tradeToAdd.soldCurrency = trade.currency;
+        tradeToAdd.amountSold = parseFloat(trade.gross_amount);
+        tradeToAdd.rate = parseFloat(trade.gross_amount) / 0;
+    } else if (trade.direction.toUpperCase() === LiquidOrderDirection.RECEIVE) {
+        tradeToAdd.soldCurrency = 'USDT';
+        tradeToAdd.boughtCurrency = trade.currency;
+        tradeToAdd.amountSold = 0;
+        tradeToAdd.rate = 0 / parseFloat(trade.gross_amount);
+        // This case here above does not work. We cannot, with current structure, create a trade with nothing sold but something bought. It should be an income (airdrop). Should we process the incomes here with the trades or import the same file again as income data file?
+    }
+    else {
+        console.info(trade);
+        throw new Error(`Error parsing ${tradeToAdd.exchange} trade.direction=${trade.direction}`);
+    }
+    tradeToAdd.ID = createID(tradeToAdd);
+    return tradeToAdd as ITrade;
+}
+
+function addFeeTrade(
+    tradeToAdd: IPartialTrade,
+    feeTrade: ILiquid,
+    rebateFeeTrade?: ILiquid,
+) : ITrade {
+    let amount = parseFloat(feeTrade.gross_amount);
+    if (rebateFeeTrade !== undefined) {
+        amount -= parseFloat(rebateFeeTrade.gross_amount);
+    }
+    tradeToAdd.boughtCurrency = 'USDT';
+    tradeToAdd.soldCurrency = feeTrade.currency;
+    tradeToAdd.amountSold = amount;
+    tradeToAdd.rate = amount / 0;
```

Comment:
Why are we dividing by zero here?

Reply:
I’m paying fee but not receiving anything for that

Reply:
I guess if we add tradeFee and tradeFeeCurrency, that would help. I’ll make a commit

Reply:
Ok so I replaced this with a trade of amount 0, rate 1, and it has a fee that is the actual amount that is paid.